src/validation.cpp

```diff
@@ -4301,9 +4305,11 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CChainState& active_ch
             }
         }
         // check level 3: check for inconsistencies during memory-only disconnect of tip blocks
-        if (nCheckLevel >= 3 && (coins.DynamicMemoryUsage() + active_chainstate.CoinsTip().DynamicMemoryUsage()) <= active_chainstate.m_coinstip_cache_size_bytes) {
+        int64_t curr_coins_usage = coins.DynamicMemoryUsage() + chainstate.CoinsTip().DynamicMemoryUsage();
+
+        if (nCheckLevel >= 3 && static_cast<unsigned long>(curr_coins_usage) <= chainstate.m_coinstip_cache_size_bytes) {
```

Comment:
In commit "refactor: rename active_chainstate in VerifyDB" (de836511d892331c0ab8d917c9cf0ca8d81537ee)

It doesn't really matter and this is fixed in the next commit, but there are three different types here: `int64_t` for `curr_coins_usage`, `unsigned long` for the cast, and `size_t` for `m_coinstip_cache_size_bytes`, and it would be nice to use `size_t` for all three from the start.